server/aesdsocket.c

```diff
@@ -0,0 +1,319 @@
+ 	
+ 	if (current_node == NULL){
+ 		syslog((LOG_USER | LOG_INFO),"Error binding socket! No socket binded, exiting program...");
+		return -1;
+ 	}
+ 	else{
+ 		syslog((LOG_USER | LOG_INFO),"Socket successfully binded!");
+ 		if (argc > 1){
+ 			if (!strcmp(argv[1],"-d")){
+ 				//we are goin demon mode
+				if ( daemon(0,1) == -1){
+					syslog((LOG_USER | LOG_INFO),"Error binding socket! No socket binded, exiting program...");
+					return -1;
+				}
+				
+ 			}
+ 		}
+ 	}
+ 	
+ 	//listen for connections
+ 	rc = listen(socket_fd, 5); //wait for connection, allow backlog of 5 waiters...
+ 	if (rc == -1){
+ 		syslog((LOG_USER | LOG_INFO),"Error listening to socket");
+ 		return -1;
+ 	}
+ 		
+ 	//Open or Create file for input data
+ 	int packetdata_fd = open("/var/tmp/aesdsocketdata", (O_RDWR | O_CREAT | O_APPEND), 0644);
+ 	if (packetdata_fd == -1){
+ 		syslog((LOG_USER | LOG_INFO),"Errror creating/opening temp data file");
+ 		return -1;
+ 	}
+	
+	//TODO: This does introduce a race condition around signal flag if it is updated between times where it is checked.
+	//It would be better to poll on a signal fd. Can try if I have time.
+ 	//forever wait for connections...
+ 	while(!signal_flag){ //this loop will define our connection scope
+ 		//reference: "https://beej.us/guide/bgnet: pg. 28"
+ 		struct sockaddr_storage client_addr; //address of the remote client connecting. Note that sockaddr_storage can fit ipv6 or v4
+ 		socklen_t client_addr_size = sizeof(client_addr);
+
+ 		int connection_fd;
+ 		char client_ip_str[INET6_ADDRSTRLEN];
+ 		const char* client_ip_res;
+ 		int connect_success = 0;
+ 		
+ 		connection_fd = accept(socket_fd, (struct sockaddr*)&client_addr, &client_addr_size);
+ 		if(connection_fd == -1){
+ 			if ((errno == EINTR) && signal_flag) {
+ 				syslog((LOG_USER | LOG_INFO),"socket accept interrupted by signal, begin clean termination...\r\n");
+ 			}
+ 			else{
+ 				syslog((LOG_USER | LOG_INFO),"Error accepting socket connection");
+ 				return -1;
+ 			}
+ 		}
+ 		else{
+ 			//successful connection!
+ 			connect_success = 1;
+ 			//reference: https://stackoverflow.com/questions/12810587/extracting-ip-address-and-port-info-from-sockaddr-storage
+ 			client_ip_res = inet_ntop(AF_INET6,&(((struct sockaddr_in6*)((struct sockaddr *)&client_addr))->sin6_addr) ,client_ip_str,sizeof(client_ip_str));
+ 			
+ 			
+ 			if (client_ip_res == NULL){
+ 				syslog((LOG_USER | LOG_INFO),"Errror extracting IP from client address");
+ 				return -1;
+ 			}
+ 		
+ 			syslog((LOG_USER | LOG_INFO),"Accepted connection from %s",client_ip_res);
+ 		}
+		
+ 		//dynamically allocate a buffer for receiving
+ 		//recv_buff = (char*)malloc(RECVBUFF_SIZE*sizeof(char));
+ 		char* recv_buff = (char*)malloc(1*sizeof(char));//receiving on char at a time will make logic easier, but will likely impact performance
+ 		//NOTE: man page says if recv buffer isn't big enough for whole msg some may be discarded, so I may want to change this
```

Comment:
Good thing to note but I think you are safe with `SOCK_STREAM`.  You would need this if you were using UDP and `SOCK_DGRAM` I believe.